kai/llm_interfacing/model_provider.py

```diff
@@ -168,6 +174,16 @@ def __init__(self, config: KaiConfigModels):
         else:
             self.llama_header = config.llama_header
 
+    def invoke(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        *,
+        stop: Optional[list[str]] = None,
+        **kwargs: Any,
+    ) -> Any:
```

Comment:
Isn't the output have to be a BaseMessage?

Reply:
You are correct! :astonished: